src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -780,14 +785,28 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
             break;
         }
 
-        // The system request the min/max window size and position
+        // Fix violations of minimum or maximum size
         case WM_GETMINMAXINFO:
         {
             // We override the returned information to remove the default limit
             // (the OS doesn't allow windows bigger than the desktop by default)
-            auto* info             = reinterpret_cast<MINMAXINFO*>(lParam);
-            info->ptMaxTrackSize.x = 50000;
-            info->ptMaxTrackSize.y = 50000;
+
+            const auto maximumSize = contentSizeToWindowSize(getMaximumSize().value_or(Vector2u(50'000, 50'000)), m_handle);
```

Comment:
I know the magic number was already there, is it just supposed to be "big enough" to exceed all realistic window sizes?

Also, variable name seems confusing, should probably be
```suggestion
            const auto maximumSize = getMaximumSize().value_or(Vector2u(50'000, 50'000));
            const auto windowSize = contentSizeToWindowSize(maximumSize, m_handle);
```

Reply:
That most definitely is for that purpose. It would be better if you use u32: :MAX size for this if you can. That would more semantically state that it is for that purpose.

Reply:
It's unfortunately not Rust, so `std::numeric_limits<unsigned int>::max()` will have to do 😉

(`unsigned int` because we're using `Vector2u` here)

Reply:
> is it just supposed to be "big enough" to exceed all realistic window sizes?

Supposedly? In the year 2023 it's a safe bet that few SFML users have 50k displays. Maybe they'll be commonplace in 10 years though ¯\\\_(ツ)\_/¯.

I tried to replace `50'000` with `std::numeric_limits<unsigned int>::max()` but I got a bunch of GL errors so I don't think we can do that. The magic numbers will have to stay.

> Also, variable name seems confusing, should probably be

Lower down I use that same pattern for `minimumSize`. It would be asymmetric to change `maximumSize` to `windowSize` when `minimumSize` is also a type of window size. I could use `maximumWindowSize` and `minimumWindowSize` if you'd like though.

Reply:
> That most definitely is for that purpose. It would be better if you we have u32::MAX size for this if you can. That would more semantically state that it is for that purpose.
> 
> @ChrisThrasher or @ShadowMarker789 what do you think? Maybe there is a windows thing that doesn't let it get too big? If so, I would think u16::MAX would be better.

As it happens, by default several fields for the maximum values of the window tracking and/or maximized values will be defaulted to the working size of the machine. 

Where the working size of the machine is defined as the total display rectangle of all combined virtual displays minus the rectangle of the taskbar if it is not auto-hiding. 

If you are testing the Win32 implementation, set a breakpoint prior to the assigning of the values of the fields of the tracking sizes - you'll see that they have prepopulated default values that are "reasonable" for the current computer environment.